tv_apps/smart-system/js/app_window.js

```diff
@@ -1667,18 +1667,17 @@
    * @param  {String} url URL.
    */
   AppWindow.prototype.modifyURLatBackground = function aw_changeURL(url) {
-    // If the app is in foreground, it's too risky to change it's
-    // URL. We'll ignore this request.
-    if (!this.isActive()) {
-      var iframe = this.browser.element;
-      // If the app is opened and it is loaded to the correct page,
-      // then th=ere is nothing to do.
-      if (iframe.src !== url) {
-        // Rewrite the URL of the app frame to the requested URL.
-        // XXX: We could ended opening URls not for the app frame
-        // in the app frame. But we don't care.
-        iframe.src = url;
-      }
+    // XXX: If the app is in foreground, it's too risky to change it's
+    // URL. We still change it because the home app is an overlay on top of any
+    // app.
+    var iframe = this.browser.element;
```

Comment:
Nice catch. This means we should create a black list to prevent this behavior on some apps. I would prefer to use the simplest code here and add the black list when we really need one. Since this API is based on IAC, no 3rd-party app can access it. We can prevent it by programmers.